app/validation/ValidatedActions.scala

```diff
@@ -25,13 +25,24 @@ abstract class ValidatedActions() extends TokenSecurity with I18nSupport {
     }
   }
 
-  def transferAgreementExistsAction(consignmentId: UUID)(f: Request[AnyContent] => Result): Action[AnyContent]
-  = secureAction.async { implicit request: Request[AnyContent] =>
+  def uploadPermitted(consignmentId: UUID)(f: Request[AnyContent] => Result): Action[AnyContent] = secureAction.async {
```

Comment:
I didn't yet because the tests in `UploadControllerSpec` cover the different paths through this method so I didn't think we needed tests for this method specifically. I'm happy to put some in but that's why I didn't originally. What do you think?

Reply:
I'd rather see separate unit tests, as don't really like relying on tests for another class to check logic from another class, can lead to issues when things get more complicated in the future.

Reply:
Yeah good point. I've added in some tests for this method and I added in a couple for the existing consignmentExists method as well. I think we're going to start using this as some kind of central validation point so the tests will be useful.